mk/build-target-determinator.bash

```diff
@@ -0,0 +1,52 @@
+#!bash
+#
+# A fairly-reliable method of determining the SPKs that need to be rebuild based on the files that
+# changed in the last commit.  Only looks at cross/* native/* and spk/* because, well, mk/* might
+# be a bit harder, and toolchains/* would require a rebuild of the world.
+#
+# I did this as a script rather than inside a mk/spksrc.spk.mk target to ensure I'm not evaluating
+# this on every run, only when needed.
+
+# test via:  git diff --stat HEAD^^^ | bash mk/build-target-determinator.bash
```

Comment:
This "test via" comment might need edits -- this is how I test but it's now driven by the delta from the most recent diff common to both master and current branch.